Dockerfiles/mlpack-dependencies-debian/Dockerfile

```diff
@@ -0,0 +1,58 @@
+#Using debian's latest image as base-image
+FROM debian:latest
+
+RUN apt-get update  && apt-get install -y aptitude && apt-get purge -y \
+    $(aptitude search '~i!~M!~prequired!~pimportant!~R~prequired! \
+    ~R~R~prequired!~R~pimportant!~R~R~pimportant!busybox!grub!initramfs-tools' \
+    | awk '{print $2}') && apt-get purge -y aptitude && \
+    apt-get autoremove -y && apt-get clean && rm -rf /usr/share/man/?? && \
+    rm -rf /usr/share/man/??_*
+
+#Installing dependencies required to run mlpack
+RUN apt-get update  && apt-get install -y --no-install-recommends wget \
+    libboost-math-dev libboost-program-options-dev libboost-test-dev \
+    libboost-serialization-dev cmake g++ binutils-dev make g++ txt2man git \
+    doxygen liblapack-dev libblas-dev libarpack2 libsuperlu-dev  && \
+    apt-get clean && rm -rf /usr/share/man/?? && rm -rf /usr/share/man/??_* && \
+    rm -rf /var/lib/apt/lists/* && rm -rf /usr/share/locale/* && \
+    rm -rf /var/cache/debconf/*-old && rm -rf /usr/share/doc/*
+
+#Installing armadillo via source-code
+ADD armadillo-code /armadillo-code
+WORKDIR armadillo-code
+RUN cmake .
+RUN make
+RUN make install
+
+#Removing setup files
+RUN rm -rf /armadillo-code
```

Comment:
It might be possible to get the container size smaller still by running all of the Armadillo configuration in one step and removing the sources there too.  To my understanding, after each `RUN` command, a Docker overlay is created with the diffs from the previous overlay.  Thus with these as separate commands, one overlay gets created with all the `armadillo-code/` sources, and then another gets created removing all of that.  So this means that you don't actually see any space savings when you run `docker images`.